README.md

```diff
@@ -4,9 +4,20 @@
 
 Maintains Maker protocol by poking oracles, liquidating vaults, managing the autoline, managing D3Ms, etc.
 
-## Overview
+Main Contracts:
 
-![Architecture](/docs/overview.png)
+- [`DssCronKeeper.sol`](/contracts/DssCronKeeper.sol)
+  - Executes pending jobs from [MakerDAO Sequencer](https://github.com/makerdao/dss-cron/)
+  - Triggers upkeep refunding when needed by calling the topup contract
+  - Chainlink Keepers Compatible
+- [`DssVestTopUp.sol`](/contracts/DssVestTopUp.sol)
+  - Checks whether upkeep refund is viable
+  - Refunds upkeep by
+    - Withdrawing accrued tokens from [MakerDAO Vest](https://github.com/makerdao/dss-vest)
+    - Using transferred tokens in case of emergency
+    - Swapping tokens for LINK
+
+[Architecture Overview](/docs/overview.png)
```

Comment:
```suggestion
Architecture Overview

```mermaid
graph TD
 A(ChainLink Keeper Network) -->|calls upkeeps| B(Keeper Registry)
 B -->|check and perform| C(DssCronKeeper)
 C -->|top up| D(DssVestTopUp)
 C -->|check jobs| E(DssCron)
 D -->|withdraw DAI| F(DssVest)
 D -->|return excess DAI| G(DaiJoin)
 D -->|check balance / add funds| B
 D -->|swap DAI for LINK| H(Uniswap)
```
```

Reply:
I think it will be easier to track and update changes of the architecture like this.

Reply:
Very cool! Thanks.